File: app/src/main/java/com/example/todoapp/MainActivity3.java
package com.example.todoapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity3 extends AppCompatActivity {
    MyRecyclerViewAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main3);

        ArrayList<String> todo = (ArrayList<String>)getIntent().getSerializableExtra("todo");

        // set up the RecyclerView
        RecyclerView recyclerView = findViewById(R.id.tv);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        adapter = new MyRecyclerViewAdapter(this, todo);
        recyclerView.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/todoapp/MainActivity3.kt
package com.example.todoapp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import java.util.*

class MainActivity3 : AppCompatActivity() {
    var adapter: MyRecyclerViewAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main3)
        val todo = intent.getSerializableExtra("todo") as ArrayList<String>?

        // set up the RecyclerView
        val recyclerView = findViewById<RecyclerView>(R.id.tv)
        recyclerView.layoutManager = LinearLayoutManager(this)
        adapter = MyRecyclerViewAdapter(this, todo!!)
        recyclerView.adapter = adapter
    }
}